### PR TITLE
Handle empty tab bar double clicks

### PR DESCRIPTION
**Pull request by lawrencecchen**
## Summary
- add a regression test for double-clicking empty trailing tab-bar space
- request a new terminal tab when the empty Bonsplit tab bar is double-clicked

## Testing
- `swift test`

---
## Summary by cubic
Double-clicking empty space in the tab bar now opens a new Terminal tab, without consuming unrelated double-clicks. Adds a regression test to lock this behavior in.

- **New Features**
  - Detects double-clicks on empty trailing tab-bar space and triggers `requestNewTab(kind: "terminal", inPane:)` when interactive; unhandled events pass through.
  - Adds `EmptyTabBarDoubleClickMonitorView` using a local event monitor.

<sup>Written for commit f6d6ab720f3f796dc10c11b258071d2ae0aae624. Summary will update on new commits.</sup>

## Summary by CodeRabbit

* **New Features**
  * Double-click in empty areas of the tab bar (trailing empty space and end drop zone) to quickly create a new terminal tab.
* **Tests**
  * Added automated test coverage verifying double-clicking empty tab-bar areas requests a new terminal tab.

### CODE REVIEW

File: Sources/Bonsplit/Internal/Views/TabBarView.swift [hunk 1]
```diff
@@ -568,6 +578,53 @@ private struct SplitActionButtonStyle: ButtonStyle {
...
+            guard event.clickCount >= 2 else { return event }
+            guard let coordinator, let view = coordinator.view, let window = view.window else { return event }
+            guard event.window === window else { return event }
+
+            let point = view.convert(event.locationInWindow, from: nil)
+            guard view.bounds.contains(point) else { return event }
+
+            coordinator.onDoubleClick?()
+            return nil
+        }
```
[the remark this answers is not in the document]
Comment (lawrencecchen) [reply Sources/Bonsplit/Internal/Views/TabBarView.swift:621]: Fixed in f6d6ab7. The monitor now returns the original event unless the tab bar is interactive and the double-click was actually handled, so inactive Bonsplit views no longer swallow the event.
